tests/Unit/RuleSet/RuleSetTest.php

```diff
@@ -305,6 +305,46 @@ public function addRuleWithSiblingNotInRuleSetAddsRuleAfterInitialRulesAndSetsVa
         self::assertGreaterThanOrEqual(0, $ruleToAdd->getColumnNumber(), 'column number not valid');
     }
 
+    /**
+     * @test
+     *
+     * @param non-empty-list<string> $initialPropertyNames
+     * @param int<0, max> $indexToRemove
+     *
+     * @dataProvider provideInitialPropertyNamesAndIndexOfOne
+     */
+    public function removeRuleRemovesRuleInSetAndKeepsOthers(array $initialPropertyNames, int $indexToRemove): void
```

Comment:
I'd like us to split this into two tests: One that tests that the rule gets removed, and another that tests that the other rules stay. (Generally, an "and" is a code smell that points to a test secretly containing two tests.)